independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -292,8 +294,9 @@ public MethodKey(MethodInfo method) {
         public int hashCode() {
             final int prime = 31;
             int result = 1;
-            result = prime * result + ((name == null) ? 0 : name.hashCode());
-            result = prime * result + ((params == null) ? 0 : params.hashCode());
+            result = prime * result + name.hashCode();
+            result = prime * result + params.hashCode();
+            result = prime * result + returnType.hashCode();
```

Comment:
Huh, so it's just a matter of including the return type in equality? The rest of your logic already handled that? Cool.

Reply:
> Huh, so it's just a matter of including the return type in equality?

When including the return type then we'll also generate delegating methods for methods declared on superinterfaces. Hopefully, it will not break anything ;-).